library/src/main/java/org/xmtp/android/library/Client.kt

```diff
@@ -289,34 +289,62 @@ class Client() {
         }
     }
 
-    // This is a V3 only feature
-    suspend fun createOrBuild(
+    private suspend fun initializeV3Client(
+        accountAddress: String,
+        clientOptions: ClientOptions,
+        signingKey: SigningKey? = null,
+    ): Client {
+        val inboxId = getOrCreateInboxId(clientOptions, accountAddress)
+
+        val (libXMTPClient, dbPath) = ffiXmtpClient(
+            clientOptions,
+            signingKey,
+            clientOptions.appContext,
+            null,
+            accountAddress,
+            inboxId
+        )
+
+        libXMTPClient?.let { client ->
+            return Client(
+                accountAddress,
+                client,
+                dbPath,
+                client.installationId().toHex(),
+                client.inboxId(),
+                clientOptions.api.env
+            )
+        } ?: throw XMTPException("Error creating V3 client: libXMTPClient is null")
+    }
+
+    // Function to create a V3 client with a signing key
+    suspend fun createV3(
         account: SigningKey,
-        options: ClientOptions,
+        options: ClientOptions? = null,
     ): Client {
         this.hasV2Client = false
-        val inboxId = getOrCreateInboxId(options, account.address)
-
+        val clientOptions = options ?: ClientOptions(enableV3 = true)
+        val accountAddress =
+            if (account.isSmartContractWallet) "eip155:${account.chainId}:${account.address.lowercase()}" else account.address.lowercase()
```

Comment:
This was the fix for the `Unknown Signer` issue. We need to create the client with the CIAP address.